src/sdk/trace/span.rs

```diff
@@ -142,14 +146,27 @@ impl Drop for SpanInner {
     /// Report span on inner drop
     fn drop(&mut self) {
         if let Some(data) = self.data.take() {
-            if let Ok(mut inner) = data.lock() {
-                if inner.end_time == inner.start_time {
-                    inner.end_time = SystemTime::now();
-                }
-                let exportable_span = Arc::new(inner.clone());
+            if let Ok(mut span_data) = data.lock().map(|mut data| data.take()) {
                 if let Some(provider) = self.tracer.provider() {
-                    for processor in provider.span_processors() {
-                        processor.on_end(exportable_span.clone())
+                    // Set end time if unset or invalid
+                    if let Some(data) = span_data.as_mut() {
+                        if data.end_time <= data.start_time {
+                            data.end_time = SystemTime::now();
+                        }
+                    }
+                    let last_processor_idx = provider.span_processors().len().saturating_sub(1);
```

Comment:
Seems like there should be a simpler way to do this, basically in the single processor/exporter case this should just move the span data, if there are multiple it should clone all but the last iteration.

Reply:
Maybe use `peek()` to determine whether it's the last element? Something like 
```rust
                    let mut it = provider.span_processors().iter().peekable();
                    while let Some(processor) = it.next(){
                        let span_data = if it.peek().is_some() {
                            // not last
                            span_data.clone()
                        } else {
                            // last
                            span_data.take()
                        };
                        if let Some(span_data) = span_data {
                            processor.on_end(span_data);
                        }
                    }
```

Reply:
Yeah that's probably cleaner, meant more the pattern of "move on last iteration, clone on others" but seems like this is the simplest way of accomplishing that.